iot_config/tests/route_service.rs

```diff
@@ -321,18 +312,40 @@ async fn assert_route_received(
     assert_eq!(&streamed_route.id, expected_id);
 }
 
+fn assert_route_result(
+    records: &[proto::RouteStreamResV1],
+    expected_action: proto::ActionV1,
+    expected_id: &str,
+) {
+    for record in records {
+        if let proto::RouteStreamResV1 {
+            action,
+            data: Some(proto::route_stream_res_v1::Data::Route(received_route)),
+            ..
+        } = record
+        {
+            assert_eq!(*action, expected_action as i32);
+            assert_eq!(&received_route.id, expected_id);
+            return;
+        }
+    }
+    panic!("expected message not found")
```

Comment:
could be more specific as to which msg is expected here and in the other instance of same msg